src/sentry/models/featureadoption.py

```diff
@@ -0,0 +1,168 @@
+manager.add(69, "notification_integration", "Notification Integration", "web", prerequisite=["first_project"])
+# TODO(ehfeng) manager.add("snooze", "Snooze Issue", "web", prerequisite=["first_event"])
+# TODO(ehfeng) manager.add("merge", "Merge Issues", "web", prerequisite=["first_event"])
+# TODO(ehfeng) manager.add("releases", "Releases", "web", prerequisite=["first_project"])
+
+# Admin UI
+manager.add(80, "sso", "SSO", "admin", prerequisite=["invite_team"])
+manager.add(81, "data_scrubbers", "Data Scrubbers", "admin", prerequisite=["first_event"])
+
+
+class FeatureAdoptionManager(BaseManager):
+    def in_cache(self, organization_id, feature_id):
+        org_key = FEATURE_ADOPTION_REDIS_KEY.format(organization_id)
+        feature_matches = []
+        with redis.clusters.get('default').map() as client:
+            feature_matches.append(client.sismember(org_key, feature_id))
+
+        return any([p.value for p in feature_matches])
+
+    def set_cache(self, organization_id, feature_id):
+        org_key = FEATURE_ADOPTION_REDIS_KEY.format(organization_id)
+        with redis.clusters.get('default').map() as client:
+            client.sadd(org_key, feature_id)
+
+    def get_all_cache(self, organization_id):
+        org_key = FEATURE_ADOPTION_REDIS_KEY.format(organization_id)
+        result = []
+        with redis.clusters.get('default').map() as client:
+            result.append(client.smembers(org_key))
+
+        return {int(x) for x in set.union(*[p.value for p in result])}
+
+    def bulk_set_cache(self, organization_id, *args):
+        org_key = FEATURE_ADOPTION_REDIS_KEY.format(organization_id)
+        with redis.clusters.get('default').map() as client:
+            client.sadd(org_key, *args)
+
+    def record(self, organization_id, feature_slug, **kwargs):
+        try:
+            feature_id = manager.get_by_slug(feature_slug).id
+        except KeyError as e:
+            logger.exception(e)
+            return
+
+        if not self.in_cache(organization_id, feature_id):
+            row, created = self.create_or_update(
+                organization_id=organization_id,
+                feature_id=feature_id,
+                complete=True)
+            self.set_cache(organization_id, feature_id)
+            return created
+
+        return False
+
+    def bulk_record(self, organization_id, feature_slugs, **kwargs):
+        try:
+            features = []
+            incomplete_feature_ids = set([manager.get_by_slug(slug).id for slug in feature_slugs]) - self.get_all_cache(organization_id)
+        except KeyError as e:
+            logger.exception(e)
+            return
+
+        for feature_id in incomplete_feature_ids:
+            features.append(FeatureAdoption(
+                organization_id=organization_id,
+                feature_id=feature_id,
+                complete=True))
+        try:
+            self.bulk_create(features)
+        except IntegrityError as e:
```

Comment:
wrapped in transaction. I assume with transactions, you need to return True? I'm following the pattern from `OrganizationOnboardingManager`